packages/bot/src/components/toolbar.jsx

```diff
@@ -118,6 +119,24 @@ const BotNameBox = ({ onBotNameTyped, file_name }) => (
     </div>
 );
 
+const saveStatusIcon = index => {
+    switch (index) {
+        case 1:
+            return <div className='loader toolbar__draft-icon' />;
+        case 2:
+            return <CheckIcon className='toolbar__draft-icon' />;
+        default:
+            return <></>;
+    }
+};
```

Comment:
Can we create `computed` value for this (e.g. `is_saving_workspace`) and update the render function of this component to check this value and render based on it?